### PR TITLE
[0.62] Use HRESULT instead of catching exceptions for WebSocket APIs

### LINKED ISSUE #4995: WinRTWebSocketResource throws on Windows API errors
Exceptions are thrown, which always show in the debugger even if they are ultimately caught and managed.

Leverage the `DEFAULT_CPPWINRT_EXCEPTIONS` to make the Windows-provided WinRT APIs return a `winrt::hresult` value instead of throwing.

### PR DESCRIPTION
**Pull request by JunielKatarn**
Backport of #5174.

Closes #4995.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6568)

### CODE REVIEW

File: vnext/ReactWindowsCore/ReactWindowsCore.vcxproj [hunk 1]
```diff
@@ -1,7 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="16.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(SolutionDir)\packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('$(SolutionDir)\packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props')" />
```
Comment (dannyvv) [inline vnext/ReactWindowsCore/ReactWindowsCore.vcxproj:3]: are we sure we want WinRT in Desktop?
@aeulitz Will this cause issues?

File: vnext/ReactWindowsCore/ReactWindowsCore-Desktop.vcxproj [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="16.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(SolutionDir)\packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('$(SolutionDir)\packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props')" />
```
Comment (dannyvv) [inline vnext/ReactWindowsCore/ReactWindowsCore-Desktop.vcxproj:3]: are we sure we want WinRT in Desktop?
@aeulitz Will this cause issues?

Comment (JunielKatarn) [reply vnext/ReactWindowsCore/ReactWindowsCore-Desktop.vcxproj:3]: We need C++/WinRT in Desktop. For instance, the WebSocket module is implemented with such APIs.

Note, C++/WinRT can be projected into "classic" Win32 apps, as long as the underlying API is available in that Windows version.

**Review by NickGerleman — APPROVED** [review]

**Review by dannyvv — APPROVED** [review]
:shipit:

File: vnext/ReactWindowsCore/ReactWindowsCore-Desktop.vcxproj (hunk 2 above)
Comment (Khalef1) [reply vnext/ReactWindowsCore/ReactWindowsCore-Desktop.vcxproj:3]: We currently support Windows 8.1 in desktop. Please make sure things work fine on that OS version.

**Review by Khalef1 — APPROVED** [review]

**Pull request merged**